xbmc/profiles/ProfileManager.cpp

```diff
@@ -194,26 +194,27 @@ bool CProfileManager::Load()
 
   if (m_profiles.empty())
   { // add the master user
-    CProfile profile("special://masterprofile/", "Master user", 0);
+    CProfile profile("special://masterprofile/", "Master user", MASTER_PROFILE_ID);
     AddProfile(profile);
   }
 
   // check the validity of the previous profile index
   if (m_lastUsedProfile >= m_profiles.size())
-    m_lastUsedProfile = 0;
+    m_lastUsedProfile = MASTER_PROFILE_ID;
 
   SetCurrentProfileId(m_lastUsedProfile);
 
   // check the validity of the auto login profile index
-  if (m_autoLoginProfile < -1 || m_autoLoginProfile >= (int)m_profiles.size())
-    m_autoLoginProfile = -1;
-  else if (m_autoLoginProfile >= 0)
+  if (m_autoLoginProfile < INVALID_PROFILE_ID ||
+      m_autoLoginProfile >= static_cast<int>(m_profiles.size()))
+    m_autoLoginProfile = INVALID_PROFILE_ID;
+  else if (m_autoLoginProfile >= MASTER_PROFILE_ID)
```

Comment:
`else if (m_autoLoginProfile > INVALID_PROFILE_ID)`

Reply:
Okay, will be changed.